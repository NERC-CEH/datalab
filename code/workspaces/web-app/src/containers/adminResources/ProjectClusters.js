import React from 'react';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { ResourceAccordion, ResourceAccordionSummary, ResourceAccordionDetails } from '../../components/common/ResourceAccordion';
import ProjectClustersContainer from '../clusters/ProjectClustersContainer';

export default function ProjectClusters(props) {
  const { project, userPermissions } = props;
  const modifyData = false;

  return (
    <ResourceAccordion defaultExpanded>
      <ResourceAccordionSummary expandIcon={<ExpandMoreIcon />}>
        <Typography variant="h6">Clusters</Typography>
      </ResourceAccordionSummary>
      <ResourceAccordionDetails>
        <ProjectClustersContainer userPermissions={userPermissions} projectKey={project.key} modifyData={modifyData} />
      </ResourceAccordionDetails>
    </ResourceAccordion>
  );
}
