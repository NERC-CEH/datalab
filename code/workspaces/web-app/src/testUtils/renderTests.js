import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
