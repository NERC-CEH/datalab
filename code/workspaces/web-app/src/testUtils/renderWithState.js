import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
