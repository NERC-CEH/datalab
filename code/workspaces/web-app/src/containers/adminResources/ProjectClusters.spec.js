import React from 'react';
import { render } from '../../testUtils/renderTests';
import ProjectClusters from './ProjectClusters';

jest.mock('../clusters/ProjectClustersContainer', () => props => (<>ProjectClustersContainer Mock {JSON.stringify(props)}</>));

describe('ProjectClusters', () => {
  const promisedUserPermissions = {
    fetching: false,
    error: null,
    value: ['some-permission'],
  };
  const project = {
    key: 'test-project',
    name: 'Test Project',
  };
  const props = {
    promisedUserPermissions,
    project,
  };

  it('renders to match snapshot passing correct props to children', () => {
    expect(render(<ProjectClusters {...props} />).container).toMatchSnapshot();
  });
});
