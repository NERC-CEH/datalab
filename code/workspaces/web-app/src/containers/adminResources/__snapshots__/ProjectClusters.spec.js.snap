// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectClusters renders to match snapshot passing correct props to children 1`] = `
<div>
  <div
    class="MuiPaper-root MuiExpansionPanel-root WithStyles(ForwardRef(ExpansionPanel))-root-1 Mui-expanded WithStyles(ForwardRef(ExpansionPanel))-expanded-2 MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiExpansionPanelSummary-root WithStyles(ForwardRef(ExpansionPanelSummary))-root-35 Mui-expanded WithStyles(ForwardRef(ExpansionPanelSummary))-expanded-37"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiExpansionPanelSummary-content WithStyles(ForwardRef(ExpansionPanelSummary))-content-38 Mui-expanded WithStyles(ForwardRef(ExpansionPanelSummary))-expanded-37"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Clusters
        </h6>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon WithStyles(ForwardRef(ExpansionPanelSummary))-expandIcon-36 Mui-expanded WithStyles(ForwardRef(ExpansionPanelSummary))-expanded-37 MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiExpansionPanelDetails-root WithStyles(ForwardRef(ExpansionPanelDetails))-root-101"
            >
              ProjectClustersContainer Mock 
              {"projectKey":"test-project","modifyData":false}
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
